### ISSUE TITLE
Won't stop saying 'QuotaExceededError'

### ISSUE THREAD

**Issue by DigitEngine** [sehugg/8bitworkshop#17]
Whenever I make changes in the code (for any platform) something keeps popping up a the top that says 'QuotaExceededError.' What is this, and how do i fix it?

**Comment by sehugg** [comment]
Go to "File >> Request Local Storage Permissions" to tell the browser to give you more quota. Did that fix it?

**Comment by DigitEngine** [comment]
No, it just says 'your browser has unlimited and persistent storage quota for this site. if you got an error while storing something, please try again.'

**Comment by sehugg** [comment]
What browser/version/OS?

**Comment by DigitEngine** [comment]
google chrome, 8bitworkshop v3.4.2, and windows 7. I don't think it has to to anything with that, because it just started happening today.

**Comment by jenh** [comment]
Can you try the following to see if it clears up the error?

With 8bitworkshop.com/v3.4.2 open, CTRL-Shift-I (or CTRL-Shift-C) to open Developer Tools, then click **Console**. At the **>** prompt, paste the following and press Enter:

```
navigator.webkitPersistentStorage.queryUsageAndQuota (
    function(usedBytes, grantedBytes) {
        console.log('We are using ', usedBytes/1024/1024, ' bytes of ', grantedBytes/1024/1024, 'MB');
    },
    function(e) { console.log('Error', e);  }
);
```

That should show how much space you're using and what Chrome's current Persistent Storage quota is set to -- you can replace "webkitPersistentStorage" with webkitTemporaryStorage" to get temporary storage (I believe 8bitworkshop sort of uses both; I haven't had any success with Chrome letting me increase temporary storage, though).

If it looks like it's capped, you can force a quota request like:

```
navigator.webkitPersistentStorage.requestQuota(1024*1024*50)
```

(where 50 is the number of MB you need -- it needs to be higher than what it's currently set to to pop up the quota request) and Chrome should pop up a request that you need to accept to increase the quota, accept it, and then you can re-run the PersistentStorage check again to see if it worked and try changing code to see if it cleared up the error.

**Comment by DigitEngine** [comment]
Both times I checked the quota, it said 'We are using 0 bytes of 0 MB.' I don't know if that means if that means that there is 0 MB left, or 0 MB used (sorry...I have no experience with this, I still don't really know what 'quota' is)

**Comment by jenh** [comment]
That means that nothing is stored and Chrome isn't letting you store anything, so the Quota Exceeded error makes sense because Chrome has your quota set to 0. ;)

Are you using an incognito window by any chance? 

If not, what happens when you paste the following into the console and press Enter:

```
navigator.webkitPersistentStorage.requestQuota(1024*1024)
```

(this will just ask for 1 MB of space as a test). Does it pop up anything and/or do you see errors in the dev console?

**Comment by DigitEngine** [comment]
It says '8bitworkshop.com wants to store files on this device', so i click allow and nothing happened.

It's like an anti-coding ghost is in my computer....
:o

**Comment by DigitEngine** [comment]
Oh jeez, chrome must be really confused, because now it doesn't automatically save

whhhaaAAAATTTTTT...?

**Comment by jenh** [comment]
It really sounds like you're in an incognito window or have disabled local storage. You may want to double-check your browser settings.

**Comment by DigitEngine** [comment]
I wasn't incognito, but it's fine. It seems to have fixed itself. Thank you both for helping!

**Comment by sehugg** [comment]
I see this still happening to people on Chrome, so I'm going to reopen.

**Comment by DigitEngine** [comment]
Okay, but could you reopen #20? I have more questions.

**Comment by DigitEngine** [comment]
never mind.

**Comment by DigitEngine** [comment]
well, there is another problem. I went to a game I'm working on but it didn't update. So I tried "Pull latest from repository" and it did nothing (as in it said it was loading, but there was no progress being made) so I deleted the repo out of the browser storage, and tried to upload it again, and it didn't work because it said there wasn't enough quota. So I tried again and said it was already in there. I canceled it, and reloaded the page. It was there, except it was the default for a new project in 8bitworkshop. What is happening here?
